core/dotnet/src/AzureIdentityAuthenticationProvider.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+using Kiota.Abstractions;
+
+namespace KiotaCore {
+    public class AzureIdentityAuthenticationProvider : IAuthenticationProvider
+    {
+        private readonly TokenCredential creds;
+        private readonly List<string> _scopes;
+        public AzureIdentityAuthenticationProvider(TokenCredential credentials, params string[] scopes)
+        {
+            creds = credentials ?? throw new ArgumentNullException(nameof(credentials));
+            if(_scopes == null)
```

Comment:
Q: does this need to check the parameter instead?
```suggestion
            if(scopes == null)
```

Reply:
good catch! thanks!